catroid/src/org/catrobat/catroid/physics/PhysicsLook.java

```diff
@@ -128,10 +133,26 @@ public void setRotation(float degrees) {
 
 	@Override
 	public void setScale(float scaleX, float scaleY) {
+		Vector2 oldScales = new Vector2(getScaleX(), getScaleY());
+		if (scaleX < 0.0f || scaleY < 0.0f) {
+			scaleX = 0.0f;
+			scaleY = 0.0f;
+		}
+
+		int scaleXComp = Math.round(scaleX * SCALEFACTOR_ACCURACY);
+		int scaleYComp = Math.round(scaleY * SCALEFACTOR_ACCURACY);
+		if (scaleXComp == Math.round(oldScales.x * SCALEFACTOR_ACCURACY) && scaleYComp == Math.round(oldScales.y * SCALEFACTOR_ACCURACY)) {
+			return;
+		}
+
+		scaleX = scaleXComp / SCALEFACTOR_ACCURACY;
+		scaleY = scaleYComp / SCALEFACTOR_ACCURACY;
 		super.setScale(scaleX, scaleY);
```

Comment:
This is not a good idea. The scale values for the look should always be the values specified by the user.